add-on/chrome/modules/LoopSidebar.jsm

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+/* Basic code borrowed and adapted from PocketAbout stuff in mozilla-central
+ */
+
+const { utils: Cu } = Components;
+const kNSXUL = "http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul";
+const SIDEBAR_WIDTH = 250;
+// This is part of the browser, so we don't need the random element.
+const BROWSER_FRAME_SCRIPT = "chrome://browser/content/content.js";
+
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+// See LOG_LEVELS in Console.jsm. Common examples: "All", "Info", "Warn", & "Error".
+const PREF_LOG_LEVEL = "loop.debug.loglevel";
+
+XPCOMUtils.defineLazyGetter(this, "log", () => {
+  let ConsoleAPI = Cu.import("resource://gre/modules/Console.jsm", {}).ConsoleAPI;
+  let consoleOptions = {
+    maxLogLevelPref: PREF_LOG_LEVEL,
+    prefix: "Loop"
+  };
+  return new ConsoleAPI(consoleOptions);
+});
+
+var LoopSidebarInternal = {
+  sidebar: null,
+
+  createSidebar: function(gBrowser) {
+    let ownerDocument = gBrowser.ownerDocument;
+    var browser = ownerDocument.getElementById("browser");
+
+    let sidebarBrowser = ownerDocument.createElementNS(kNSXUL, "browser");
+    sidebarBrowser.setAttribute("id", "loop-side-browser");
+    sidebarBrowser.setAttribute("disablehistory", "true");
+    sidebarBrowser.setAttribute("disableglobalhistory", "true");
+    sidebarBrowser.setAttribute("frameType", "loop");
+    sidebarBrowser.setAttribute("messagemanagergroup", "loop");
+    sidebarBrowser.setAttribute("message", "true");
+    sidebarBrowser.setAttribute("remote", "true");
+
+    sidebarBrowser.setAttribute("type", "content");
+
+    this.sidebar = sidebarBrowser;
+    this.sidebar.width = SIDEBAR_WIDTH;
+
+    browser.appendChild(sidebarBrowser);
+  },
+
+  loadSidebar: function(token) {
+    log.info("loadSidebar called:", token, this.sidebar);
+    let url = "about:loopconversation#" + token;
+    Services.perms.add(Services.io.newURI(url, null, null), "camera",
+                     Services.perms.ALLOW_ACTION, Services.perms.EXPIRE_SESSION);
+    this.sidebar.setAttribute("src", url);
+  }
+};
+
+this.LoopSidebar = Object.freeze({
+  init: function(window) {
+    // This loads the generic browser content script into our special "loop"
+    // message manager for the sidebar.
+    let loopMessageManager = window.getGroupMessageManager("loop");
+    loopMessageManager.loadFrameScript(BROWSER_FRAME_SCRIPT, true);
+  },
+
+  createSidebar: function(gBrowser) {
+    LoopSidebarInternal.createSidebar(gBrowser);
+  },
+
+  loadSidebar: function(token) {
```

Comment:
Having looked at this a bit, I think we could pretty easily combine `createSidebar` and `loadSidebar`. There's no reason to create one without having loaded it first.

Also, we could move the `loadFrameScript` code from init() into the `loadSidebar` function - we don't need that frame script being able to be loaded until just before we have a sidebar.